src/shared/filesystemUtilities.ts

```diff
@@ -156,6 +156,35 @@ export function isInDirectory(d: string, p: string): boolean {
     })
 }
 
+/**
+ *
+ * @returns file distance between fileA and fileB
+ * For example:
+ * The file distance between A/B/C.java and A/B/D.java is 0
+ * The file distance between A/B/C.java and A/D.java is 1
+ */
+export function getFileDistance(fileA: string, fileB: string): number {
+    let filePathA = pathutils.normalize(fileA).split('/')
+    filePathA = filePathA.slice(0, filePathA.length - 1)
+
+    let filePathB = pathutils.normalize(fileB).split('/')
+    filePathB = filePathB.slice(0, filePathB.length - 1)
+
+    let i = 0
+    while (i < Math.min(filePathA.length, filePathB.length)) {
+        const dir1 = filePathA[i].toLowerCase()
+        const dir2 = filePathB[i].toLowerCase()
```

Comment:
This would produce inaccurate results for case-sensitive file systems

Reply:
I see, let me update this part. @JadenSimon I thought what Justin mean here is to compare the string case-insensitively https://github.com/aws/aws-toolkit-vscode/pull/3638#discussion_r1252293259, so I added toLowerCase()